### DISCUSSION TITLE
Getting notified when auth is lost

### DISCUSSION THREAD

**Discussion by iandees** [teslamate-org/teslamate#2304] [Q&A] [answered]
I noticed that my charge last night wasn't showing up on Grafana, and when I went to Teslamate it was asking me for login. Looks like Tesla removed authorization for my OAuth tokens after a timeout:

```
Jan 11 19:22:42 heat teslamate[2481275]: 2022-01-11 19:22:42.732 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/xxx/vehicle_data -> 408 (9206.069 ms)
Jan 11 19:22:42 heat teslamate[2481275]: 2022-01-11 19:22:42.732 [warn] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
Jan 11 19:22:42 heat teslamate[2481275]: 2022-01-11 19:22:42.732 car_id=1 [error] Error / :unknown
```

Is there any way to have Teslamate notify me when it loses access to the API or the credentials expire so that I miss as little data as possible?

**Comment by mlrdev** [comment]
Hi, think you've asked the right questions ahead of time.
Recently Tesla seemed to enforce HTTP/2 and TLS 1.3 for the auth request, which broke the auth flow for all.
A fix was provided via https://github.com/teslamate-org/teslamate/pull/5406, still your question remains.

Do you have any custom solution locally on your setup to notify _you_, which could be made generally available eventually?

**Reply by JakobLichterfeld** [reply, marked as answer]
Healthcheck was reporting it correctly, that's why we spotted it, and we were able to fix it with v4.0.0 and the next day the next issue with V4.0.1.
Related to: #4782, especially https://github.com/teslamate-org/teslamate/discussions/4782#discussioncomment-13434566

**Reply by timguy99** [reply]
There is a healthcheck for such things somewhere?

**Reply by JakobLichterfeld** [reply]
Yes, https://github.com/teslamate-org/teslamate/blob/a115cc9989fc47e9c3010d2f994ad79e46aed8e1/lib/teslamate_web/health_check.ex#L6 and it is [shown in Dashboard if failed](https://github.com/teslamate-org/teslamate/blob/a115cc9989fc47e9c3010d2f994ad79e46aed8e1/lib/teslamate_web/live/car_live/summary.html.heex#L200), that's where most of the users see it. This can indicate Tesla servers down, connection errors or like last weekend, auth tokens failed to refresh.

**Reply by timguy99** [reply]
What I meant was, is a health endpoint available to query. If so I'm going to setup a zabbix check against my teslamate instance to alert me when it is NOT healthy. I'm guessing by the above I can hit it at http://tesla-instance:4000/health_check

**Reply by JakobLichterfeld** [reply]
For sure, http response status, like said here

> especially [#4782 (comment)](https://github.com/teslamate-org/teslamate/discussions/4782#discussioncomment-13434566)

**Reply by sp991** [reply]
> What I meant was, is a health endpoint available to query. If so I'm going to setup a zabbix check against my teslamate instance to alert me when it is NOT healthy. I'm guessing by the above I can hit it at http://tesla-instance:4000/health_check

Were you able to setup zabbix to check the health? I'm not familiar with this application and would like to set it up. I've already installed zabbix, created the host and added an http agent item but I can't figure out how to setup the agent properly :(

I would love some guidance if you don't mind

**Reply by JakobLichterfeld** [reply]
You can use whatever you want UptimeKuma for example. No agent needed in any case.
